src/digest.rs

```diff
@@ -448,8 +447,66 @@ pub extern fn SHA512_4(out: *mut u8, out_len: c::size_t,
     polyfill::slice::fill_from_slice(out, digest);
 }
 
+unsafe extern fn sha1_block_data_order(state: *mut u64, data: *const u8, num: c::size_t) {
```

Comment:
Please add some documentation about the speed-for-size/simplicity trade-off we are making here. The main purpose in retaining SHA-1 is to support legacy protocols and OCSP, none of which need a fast SHA-1 implementation. Also, please add a note that, unlike SHA-256, SHA-384, and SHA-512, there are no primary assembly language implementations.

Finally, could you please put this in a separate file, following the example of `ring::aead::chacha20_poly1305`. For the license header, please use the ISC license header used in this file, but make sure the Copyright line of the file contains the copyright holder's name.